tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -372,11 +266,153 @@ public:
+        fNElements(nElements),
+        fCompressionSettings(compressionSettings),
+        fIsSuppressed(suppressed)
    {
-      return fColumnRanges.find(physicalId) != fColumnRanges.end();
    }
-   std::uint64_t GetNBytesOnStorage() const;
+
+   ROOT::DescriptorId_t GetPhysicalColumnId() const { return fPhysicalColumnId; }
+   void SetPhysicalColumnId(ROOT::DescriptorId_t id) { fPhysicalColumnId = id; }
+
+   ROOT::NTupleSize_t GetFirstElementIndex() const { return fFirstElementIndex; }
+   void SetFirstElementIndex(ROOT::NTupleSize_t idx) { fFirstElementIndex = idx; }
+   void IncrFirstElementIndex(ROOT::NTupleSize_t by) { fFirstElementIndex += by; }
+
+   ROOT::NTupleSize_t GetNElements() const { return fNElements; }
+   void SetNElements(ROOT::NTupleSize_t n) { fNElements = n; }
+   void IncrNElements(ROOT::NTupleSize_t by) { fNElements += by; }
+
+   std::optional<std::uint32_t> GetCompressionSettings() const { return fCompressionSettings; }
+   void SetCompressionSettings(std::optional<std::uint32_t> comp) { fCompressionSettings = comp; }
+
+   bool IsSuppressed() const { return fIsSuppressed; }
+   void SetSuppressed(bool suppressed) { fIsSuppressed = suppressed; }
+
+   bool operator==(const RColumnRange &other) const
+   {
+      return fPhysicalColumnId == other.fPhysicalColumnId && fFirstElementIndex == other.fFirstElementIndex &&
+             fNElements == other.fNElements && fCompressionSettings == other.fCompressionSettings &&
+             fIsSuppressed == other.fIsSuppressed;
+   }
+
+   bool Contains(ROOT::NTupleSize_t index) const
+   {
+      return (fFirstElementIndex <= index && (fFirstElementIndex + fNElements) > index);
+   }
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RClusterDescriptor::RPageRange
+\ingroup NTuple
+\brief Records the partition of data into pages for a particular column in a particular cluster
+*/
+// clang-format on
+class RClusterDescriptor::RPageRange {
+   friend class Internal::RClusterDescriptorBuilder;
+   /// Extend this RPageRange to fit the given RColumnRange, i.e. prepend as many synthetic RPageInfos as needed to
+   /// cover the range in `columnRange`. `RPageInfo`s are constructed to contain as many elements of type `element`
+   /// given a page size limit of `pageSize` (in bytes); the locator for the referenced pages is `kTypePageZero`.
+   /// This function is used to make up `RPageRange`s for clusters that contain deferred columns.
+   /// \return The number of column elements covered by the synthesized RPageInfos
+   std::size_t ExtendToFitColumnRange(const RColumnRange &columnRange, const Internal::RColumnElementBase &element,
+                                      std::size_t pageSize);
+
+   /// Has the same length than fPageInfos and stores the sum of the number of elements of all the pages
+   /// up to and including a given index. Used for binary search in Find().
+   std::vector<ROOT::NTupleSize_t> fCumulativeNElements;
```

Comment:
A curiosity/question: how deep do we want to go with changing fields to getters and setters? Since these inner classes are public, should they have them as well?